het-sites.py

```diff
@@ -22,20 +24,123 @@
 
 length = len(reads[0])
 width = int(log10(length)) + 1
-total_heterogeneious = 0
-
-for index in range(length):
-    counts = Counter()
-    for read in reads:
-        # Do not take gaps into account.
-        if read.sequence[index] != '-':    
-            counts[read.sequence[index]] += 1
-
-    if len(counts) > 1:
-        if (counts.most_common(1)[0][1])/sum(counts.values()) < args.homogen:
-            # The site is not homogenous = more than one nucleotide present.       
-            print('Site %*d has nucleotides %s' % (width, index + 1, str(counts)[9:-2]))
-            total_heterogeneious += 1
-
-print('The total of heterogeneious sites is %i.' % (total_heterogeneious))
+
+def hetsites():
+    total_heterogeneious = 0
+
+    for index in range(length):
+        counts = Counter()
+
+        for read in reads:
+            if read.sequence[index] != '-':
+            # Do not take gaps into account.   
+                counts[read.sequence[index]] += 1
+
+        if len(counts) > 1:
```

Comment:
This can also be written as `if counts:`.  A dictionary (and a list and a string and a set) is considered "true" if it is non-empty.

Reply:
Done